src/main/java/appeng/client/gui/AEBaseGui.java

```diff
@@ -829,7 +851,7 @@ private void drawSlot( final Slot s )
 				this.zLevel = 0.0F;
 				itemRender.zLevel = 0.0F;
 
-				this.aeRenderItem.setAeStack( ( (SlotME) s ).getAEStack() );
+                this.aeRenderItem.setAeStack( s instanceof SlotME ? ( (SlotME) s ).getAEStack() : ( (SlotFake) s ).getAEStack() );
```

Comment:
maybe introduce an interface instead with a `getAEStack` method? like IAESlot, all these checks don't improve readability

Reply:
This might be a good approach, but then other slots would also need to implement that method, and other slots don't need AEItemStack.

Reply:
Why they would need it if it specifically marks slots that do have this method. Wel, maybe the name should be more specific

Reply:
Well, my solution now is to wrap the operation of getting AEItemStack inside a single method.